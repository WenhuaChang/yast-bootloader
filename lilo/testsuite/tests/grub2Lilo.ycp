/**
 * File:
 *  lilo2Grub.ycp
 *
 * Module:
 *  Bootloader configurator
 *
 * Summary:
 *  Translating device names from UNIX to grub
 *
 * Authors:
 *  Jiri Srain <jsrain@suse.cz>
 *
 * $Id$
 *
 */

{
    include "testsuite.ycp";
    import "Pkg";
    map IREAD = $["target" : $[
            "string" : "root (hd0,0)\ninstall /grub/stage1 d (hd0) /grub/stage2 0x8000 (hd0,0)/grub/menu.lst\nquit",
            "tmpdir" : "/tmp"
        ], "etc" : $[
            "fstab" : [],
            "install_inf" : $[
                "Locale" : "en_EN",
                "InstMode" : "cd"
            ],
	    "mtab" : []
	], "probe" : $[
            "disk" : [],
            "boot_arch" : "grub",
            "architecture" : "i386",
            "has_apm" : true,
            "has_pcmcia" : false,
            "has_smp" : false,
            "system" : [],
            "floppy" : $[
                "manual" : []
            ], "cdrom" : $[
                "manual" : []
            ]
        ], "sysconfig" : $[
            "console" : $[
                "CONSOLE_FONT" : "lat2-16.psfu",
                "CONSOLE_SCREENMAP" : "trivial",
                "CONSOLE_UNICODEMAP" : "",
                "CONSOLE_MAGIC" : "(K",
                "CONSOLE_ENCODING" : "ISO-8859-2"
            ]
        ], "md" : $[
	    "config" : $[]
	], "lvm" : $[
	    "vg" : $[],
	    "pv" : $[]
	], "proc" : $[
	    "swaps" : [],
	    "mounts" : []
	]
    ];
    TESTSUITE_INIT ([IREAD, $[], $[]], 0);
    include "lilo/routines.ycp";

    map EXEC = $["target" : $["bash_output": $["stdout" : "/dev/hda"]]];

    TEST (``(grub2Lilo ("(hd0)")), [$[], $[], EXEC], 0);
    TEST (``(grub2Lilo ("(hd0,0)")), [$[], $[], EXEC], 0);
    TEST (``(grub2Lilo ("(hd0,11)")), [$[], $[], EXEC], 0);

    EXEC = $["target" : $["bash_output": $["stdout" : "/dev/ida/c0d0"]]];

    TEST (``(grub2Lilo ("(hd0)")), [$[], $[], EXEC], 0);
    TEST (``(grub2Lilo ("(hd0,0)")), [$[], $[], EXEC], 0);
    TEST (``(grub2Lilo ("(hd0,11)")), [$[], $[], EXEC], 0);

    EXEC = $["target" : $["bash_output": $["stdout" : "/dev/fd0"]]];

    TEST (``(grub2Lilo ("(fd0)")), [$[], $[], EXEC], 0);

}

