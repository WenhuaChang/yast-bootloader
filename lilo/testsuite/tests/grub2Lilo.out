Read	.probe.architecture "i386"
Read	.probe.has_apm true
Read	.probe.has_pcmcia false
Read	.probe.has_smp false
Read	.probe.system []
Read	.target.tmpdir "/tmp"
Read	.target.size "/etc/install.inf" 0
Read	.etc.install_inf.Sourcemounted 0
Read	.etc.install_inf.AutoYaST 0
Read	.etc.install_inf.Demo 0
Read	.etc.install_inf.Textmode 0
Read	.etc.install_inf.Manual 0
Read	.etc.install_inf.Console 0
Read	.etc.install_inf.Braille 0
Read	.etc.install_inf.VNC 0
Read	.etc.install_inf.UseSSH 0
Read	.etc.install_inf.Locale "en_EN"
Read	.etc.install_inf.InstMode "cd"
Read	.sysconfig.console.CONSOLE_FONT "lat2-16.psfu"
Read	.sysconfig.console.CONSOLE_SCREENMAP "trivial"
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC "(K"
Read	.sysconfig.console.CONSOLE_ENCODING "ISO-8859-2"
Execute	.target.bash "/usr/bin/setfont" 0
Read	.probe.floppy.manual []
Read	.probe.cdrom.manual []
Read	.probe.boot_arch "grub"
Read	.probe.disk []
Read	.md.config $[]
Read	.lvm.vg $[]
Read	.lvm.pv $[]
Read	.target.size "/etc/cryptotab" 0
Read	.proc.swaps []
Write	.dumpto.tmp.targetMap_i $[] true
Read	.proc.mounts []
Read	.proc.swaps []
Read	.etc.mtab []
Read	.target.size "/etc/fstab" 0
Read	.target.size "/etc/cryptotab" 0
Execute	.target.bash_output "awk '\n/(hd0)/ {printf $2; next;}\n// {next;}' /boot/grub/device.map" $["stdout":"/dev/hda"]
Return	/dev/hda
Execute	.target.bash_output "awk '\n/(hd0)/ {printf $2; next;}\n// {next;}' /boot/grub/device.map" $["stdout":"/dev/hda"]
Return	/dev/hda1
Execute	.target.bash_output "awk '\n/(hd0)/ {printf $2; next;}\n// {next;}' /boot/grub/device.map" $["stdout":"/dev/hda"]
Return	/dev/hda12
Execute	.target.bash_output "awk '\n/(hd0)/ {printf $2; next;}\n// {next;}' /boot/grub/device.map" $["stdout":"/dev/ida/c0d0"]
Return	/dev/ida/c0d0
Execute	.target.bash_output "awk '\n/(hd0)/ {printf $2; next;}\n// {next;}' /boot/grub/device.map" $["stdout":"/dev/ida/c0d0"]
Return	/dev/ida/c0d0p1
Execute	.target.bash_output "awk '\n/(hd0)/ {printf $2; next;}\n// {next;}' /boot/grub/device.map" $["stdout":"/dev/ida/c0d0"]
Return	/dev/ida/c0d0p12
Execute	.target.bash_output "awk '\n/(fd0)/ {printf $2; next;}\n// {next;}' /boot/grub/device.map" $["stdout":"/dev/fd0"]
Return	/dev/fd0
